### PR TITLE
fix subgraph and create geyser task

### PR DESCRIPTION
**Pull request by jx3yang**
## Description

Existing event handler for geyser creation in the subgraph and `create-geyser` hardhat task were broken.

### `create-geyser` task changes:
Deployment and initialization of a geyser must happen at separate steps.

Due to the way the subgraph and contracts work, creating a geyser requires a very specific order. Particularly, for a `Geyser` entity to be indexed in the subgraph, the event `GeyserCreated` *must* be emitted *after* the geyser has been registered to the `GeyserRegistry`, because [a new data source will not process historical events](https://thegraph.com/docs/define-a-subgraph#instantiating-a-data-source-template) . However, a geyser must be deployed in order to be registered to the registry.

Since `Geyser` is an upgradeable contract, we can execute the following in order:
- Deploy the geyser contract + proxy, without calling the initializer
- Register the geyser to the registry
- Initialize the geyser, which will then emit `GeyserCreated`

Then, the geyser will be indexed in the subgraph as required.

### Subgraph changes:
When a geyser entity is first created, its non-derived array fields need to be set to empty arrays initially.
Additionally, objects in `rewardPoolBalances` should be derived

## Testing

- Started local network with `yarn hardhat node` and `yarn hardhat deploy --network localhost --no-verify --mock`
- Started local graph-node with `docker-compose up` (under graph-node/docker) and `yarn create-local && yarn deploy-local` (under subgraph/)
- Ran the task 
```
yarn hardhat create-geyser \
--staking-token {address of MockERC20} \
--reward-token {address of MockAmpl} \
--floor 10 \
--ceiling 20 \
--time 100 \
--network localhost
```
- Queried `http://localhost:8000/subgraphs/name/thegostep/ampleforth-geyser-v2` to make sure that the geyser was indexed

### CODE REVIEW

**Review by surudhb — APPROVED** [review]
LGTM!

File: subgraph/src/geyser.ts [hunk 1]
```diff
@@ -89,10 +98,10 @@ export function handleGeyserCreated(event: GeyserCreated): void {
   entity.scalingCeiling = geyserData.rewardScaling.ceiling
   entity.scalingTime = geyserData.rewardScaling.time
   entity.status = 'Online'
+  entity.bonusTokens = []
+  entity.rewardPoolBalances = []
```
Comment (jx3yang) [inline subgraph/src/geyser.ts]: These are explicitly needed when a geyser entity is first created / initialized. For subsequent updates on an existing geyser entity, the type guarantees from the schema will apply

File: subgraph/src/geyser.ts [hunk 2]
```diff
@@ -89,10 +98,9 @@ export function handleGeyserCreated(event: GeyserCreated): void {
   entity.scalingCeiling = geyserData.rewardScaling.ceiling
   entity.scalingTime = geyserData.rewardScaling.time
   entity.status = 'Online'
+  entity.bonusTokens = []
```
Comment (jx3yang) [inline subgraph/src/geyser.ts:101]: Need to be initialized initially since it's a non-derived array field

**Review by surudhb — APPROVED** [review]
LGTM!

**Pull request merged**